tests/unittests/CoreGraphics.Drawing/CGPathDrawingTests.cpp

```diff
@@ -249,3 +249,192 @@ DISABLED_DRAW_TEST_F(CGPath, GetBoundingBox, UIKitMimicTest) {
 
     CGPathRelease(thepath);
 }
+
+static CGRect applyBounds;
+
+static void CGPathApplyCallback(void* context, const CGPathElement* element) {
+    CGPoint* points = element->points;
+    CGContextMoveToPoint((CGContextRef)context,
+                         applyBounds.origin.x + applyBounds.size.width / 2,
+                         applyBounds.origin.y + applyBounds.size.height / 2);
+    CGContextAddLineToPoint((CGContextRef)context, points[0].x, points[0].y);
+    CGContextStrokePath((CGContextRef)context);
+}
+
+DISABLED_DRAW_TEST_F(CGPath, PathApplyDraw, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    applyBounds = GetDrawingBounds();
+    CGFloat width = applyBounds.size.width;
+    CGFloat height = applyBounds.size.height;
+    CGFloat xstart = applyBounds.origin.x;
+    CGFloat ystart = applyBounds.origin.y;
+
+    CGMutablePathRef thepath = CGPathCreateMutable();
+    CGPathMoveToPoint(thepath, NULL, xstart + .5 * width, ystart + .1 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .35 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .1 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .3 * width, ystart + .5 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .25 * width, ystart + .9 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .5 * width, ystart + .7 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .75 * width, ystart + .9 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .7 * width, ystart + .5 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .9 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .65 * width, ystart + .3 * height);
+    CGPathAddLineToPoint(thepath, NULL, xstart + .5 * width, ystart + .1 * height);
+    CGPathCloseSubpath(thepath);
+
+    CGPathApply(thepath, context, CGPathApplyCallback);
+
+    CGContextAddPath(context, thepath);
+    CGContextStrokePath(context);
+
+    CGPathRelease(thepath);
+}
+
+static void CGPathControlPointCallback(void* context, const CGPathElement* element) {
+    CGFloat colorComponents[] = { 0, 0, 0, 1 };
+
+    switch (element->type) {
+        case kCGPathElementMoveToPoint:
+            colorComponents[0] = 1;
+            colorComponents[1] = 0;
+            colorComponents[2] = 0;
+            CGContextSetRGBStrokeColor((CGContextRef)context,
+                                       colorComponents[0],
+                                       colorComponents[1],
+                                       colorComponents[2],
+                                       colorComponents[3]);
+
+            CGContextStrokeRect((CGContextRef)context, CGRectMake(element->points[0].x - 2.5, element->points[0].y - 2.5, 4, 4));
+            break;
```

Comment:
also, it might be better to do point - 2.5, + 5 instead of + 4 (so the center of the rect would be the point)

---
In reply to: [91633249](https://github.com/Microsoft/WinObjC/pull/1507#discussion_r91633249) [](ancestors = 91633249)